addons/controllers/suite_test.go

```diff
@@ -240,3 +257,126 @@ var _ = AfterSuite(func() {
 	err := testEnv.Stop()
 	Expect(err).ToNot(HaveOccurred())
 })
+
+// setupWebhooks configures certs/keys and sets up the webhooks with mgr
+func setupWebhooks(ctx context.Context, mgr manager.Manager, setupLog logr.Logger) {
+	var (
+		f   *os.File
+		err error
+	)
+	// Setup TLS certs and keys
+	setupCertsAndKeysForWebhooks()
+
+	// Apply the webhooks configuration
+	webhooksFile := "webhooks/manifests.yaml"
+	f, err = os.Open(webhooksFile)
+	Expect(err).ToNot(HaveOccurred())
+	defer f.Close()
+	err = testutil.CreateResources(f, cfg, dynamicClient)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Update the caBundle in validating and mutating webhooks
+	k8sConfig := mgr.GetConfig()
+	client, err := kubernetes.NewForConfig(k8sConfig)
+	validatingWebhookConf, err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().
```

Comment:
Could't no you just call  
``` 
addons/testutils/test_helpers.go.CreateResources(addons/controllers/webhooks/manifests.yaml)) 
```

That way this test not only validates the webhooks but also the format and functionality of the manifest.yml files. 

then if you really want to, you  cabundle by doing something like:
```
	allValidatingWebhookConfigurations, err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().List(ctx, metav1.ListOptions{LabelSelector: labelSelector})
	if err != nil {
		return err
	}
	for idx := range allValidatingWebhookConfigurations.Items {
		configName := allValidatingWebhookConfigurations.Items[idx].Name
		validatingWebhookConfiguration, err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Get(ctx, configName, metav1.GetOptions{})
		if err != nil {
			return err
		}

		for idx := range validatingWebhookConfiguration.Webhooks {
			validatingWebhookConfiguration.Webhooks[idx].ClientConfig.CABundle = secret.Data[resources.CACert]
		}
		if _, err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Update(ctx, validatingWebhookConfiguration, metav1.UpdateOptions{}); err != nil {
			return fmt.Errorf("error updating CA cert of ValidatingWebhookConfiguration %s: %v", configName, err)
		}
	}
```

Reply:
take a look at https://github.com/vmware-tanzu/tanzu-framework/pull/1740/files